### PR TITLE
Add a GH action stage to send coverage reports to Codecov

### LINKED ISSUE #263: Add codecov for code coverage
It would be great to check project's health taking a look into code coverage reports.

We all know that code coverage by itself does not mean the code has quality, but can give us an initial sense of it, and a place to look at when writing tests for uncovered places.

[Codecov](https://codecov.io/) is a great service, free for Open Source projects. It has Github Actions integration, so I see it as a convenient service to use. If added, we can add a badge at main's README file.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds a new stage for codecov in the main GH action pipeline.

To generate the test coverage, we are enabling it in the stage for the `gotestsum` execution, adding the `-coverprofile=cover.txt` flag. (this file is ignored in git)

## Why is it important?
It will allow us to share the code coverage for this project with the community, also providing an initial direction on where to improve the tests.

## Related issues 
- Closes #263

### CODE REVIEW

**Comment by mdelapenya** [comment]
I realised that every PR from a fork will fail this new stage because of the github action not being able to interact with the free layer of codecov from a fork.

If we want to enable it, it would be needed to add a GH secret. @gianarb, thoughts?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,4 +47,10 @@ jobs:
...
 
       - name: gotestsum
-        run: gotestsum --format short-verbose ./...
+        run: gotestsum --format short-verbose -- -coverprofile=cover.txt ./...
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v1.0.2
+        with:
+          file: ./cover.txt
+          name: testcontainers-go-${{ matrix.go-version }}
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:56]:
```suggestion
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          file: ./cover.txt
          name: testcontainers-go-${{ matrix.go-version }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,4 +47,10 @@ jobs:
         run: go vet ./...
 
       - name: gotestsum
-        run: gotestsum --format short-verbose ./...
+        run: gotestsum --format short-verbose -- -coverprofile=cover.txt ./...
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v1.0.2
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:53]:
```suggestion
        uses: codecov/codecov-action@v1
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mdelapenya) [reply .github/workflows/ci.yml:56]: Mmm, it was fixed with 0913f1b344a2284c2a3922a9454fdcb1711d1710, which uses the latest GH action

**Review by gianarb — APPROVED** [review]

**Comment by gianarb** [comment]
Good! As soon as the testcontainers org will approve codecov I will merge this one. Thanks

**Comment by gianarb** [comment]
@mdelapenya do you mind to rebase? :) Thanks a lot!

**Comment by mdelapenya** [comment]
Done! I merged branch with upstream

**Comment by gianarb** [comment]
Thanks a lot!!

**Comment by mdelapenya** [comment]
Working properly!! https://github.com/testcontainers/testcontainers-go/pull/288#issuecomment-785624399

**Pull request merged**